app/src/main/java/com/example/wearableapp/presentation/ui/ExportHeartRateActivity.kt

```diff
@@ -0,0 +1,109 @@
+package com.example.wearableapp.presentation.ui
+
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.Observer
+import com.example.wearableapp.R
+import com.example.wearableapp.databinding.ActivityExportHeartRateBinding
+import com.example.wearableapp.presentation.utils.CheckPermission
+import com.example.wearableapp.presentation.utils.Constants
+import com.example.wearableapp.presentation.viewmodel.ExportHeartRateViewModel
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class ExportHeartRateActivity : AppCompatActivity(), View.OnClickListener {
+
+    val TAG = ExportHeartRateActivity::class.simpleName
+    private lateinit var binding: ActivityExportHeartRateBinding
+    private val viewModel by viewModel<ExportHeartRateViewModel>()
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        binding = ActivityExportHeartRateBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        setListeners()
+    }
+
+    private fun setListeners() {
+        binding.tvHeartExConfirmId.setOnClickListener(this)
+    }
+
+    override fun onClick(v: View?) {
```

Comment:
You can set listeners for every element with binding like this `binding.tvHeartExConfirmId.setOnClickListener { code }`

it's much cleaner this way

Reply:
Resolved